src/file/file.model.ts

```diff
@@ -81,7 +77,7 @@ fileSchema.virtual('fullExtension')
 
 fileSchema.pre('save', async function (next: NextFunction) {
   const existingFile = await fileModel.findOne({ name: (<any>this).name, parent: (<any>this).parent, ownerID: (<any>this).ownerID });
-  if (existingFile && !existingFile.deleted) {
+  if (existingFile) {
```

Comment:
If you don't have `deleted` field anymore then you should make a unique index of `name`, `parent` and `ownerID`.

Reply:
done

Reply:
@omrishtam any comments or is it resolved?

Reply:
I meant that you should remove the `pre` hook, there's no need for it if you can just make an index.

Reply:
hmm.. you are right, but I still want to give a good error response (which I started to implement on the hotfix branch I created today). So I guess I'll delete the pre on this branch and handle it on the other.
TODO on sunday.

Reply:
removed